### DISCUSSION TITLE
Dynamic checkboxes

### DISCUSSION THREAD

**Discussion by Krelix1** [react-hook-form/react-hook-form#8490] [Q&A] [answered]
### Steps to reproduce
1. Click `Add checkbox`
2. Click on any checkbox twice
3. isDirty is not working

### Description
I have a case similar to that I provided in the Sandbox.
I have an infinity loader in the list of checkboxes. When new checkboxes is fetched, I call `reset` with new `defaultValues`. The goal is to keep already checked checkboxes ( to do this I provide a `keepValues / keepDirtyValues` field ). But as you can see something  go wrong and `isDirty` is breaking. Is this a bug or is there another way to do this?

**Comment by bluebill1049** [comment, marked as answer]
just like what i mentioned in the issue, with array as string order does matter, because we are doing a deep equal.

**Reply by Krelix1** [reply]
How can I use object as a default value with checkboxes? onChange returns string instead of boolean. 
For example:
I have/want:
`
{
  [id]: true/false,
  [id]: true/false
}
`
I get:
`
 {
   [id]: [id],
   [id]: [id]
 }
`

**Reply by bluebill1049** [reply]
you can build a controlled component wrapped with `controller` or hook it up `useController`. That's one of the enormous benefits of a controlled component, the component state hosts your input.
